src/main/java/io/github/cottonmc/cotton_scripting/impl/js/JavascriptParser.java

```diff
@@ -0,0 +1,51 @@
+package io.github.cottonmc.cotton_scripting.impl.js;
+
+import io.github.cottonmc.cotton_scripting.ExecutableScript;
+import io.github.cottonmc.cotton_scripting.ScriptParser;
+import io.github.cottonmc.cotton_scripting.api.InvocableScript;
+import jdk.nashorn.api.scripting.NashornScriptEngineFactory;
+import net.minecraft.util.Identifier;
+
+import javax.script.Invocable;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+import java.util.Optional;
+
+public class JavascriptParser implements ScriptParser {
+
+    private final NashornScriptEngineFactory nashornScriptEngineFactory = new NashornScriptEngineFactory();
+
+    @Override
+    public Optional<ExecutableScript> parse(String script){
+        ScriptEngine engine = nashornScriptEngineFactory.getScriptEngine(new JavascriptClassFilter());
+
+        try {
+            //remove pesky things that the user can use maliciously.
+            engine.eval("exit = undefined");
+            engine.eval("load = undefined");
+            engine.eval("Java = undefined");
+
+            engine.eval(script);
+            if (engine instanceof Invocable) {
+                Object main = engine.get("main");
+                //if there is no "main" method, then we don't return the script. It is mandatory.
```

Comment:
I don't think this is a good or necessary idea, since we can just using bindings.

Reply:
The idea here is so that I get something that I can call from the context, that is surely there.

The other option is to send in an object that is for event handling, where you bind a function from code (instead of script tagging)
```
var FunctionApi = ScriptAPI.include("functionapi","event_binding")

function handlerMethod(){
//....
}

FunctionApi.bindEvent(handlerMethod,identifier...)```

Reply:
And now I see what you mean with the binding.

Reply:
Yeah, we can pass in an object for the context at eval-time, instead of invoke time. It's definitely still a good idea to allow a script to have invocable functions for bound events. Right now, I do this in SandScript by [invoking specifically-named methods](https://github.com/Boundarybreaker/SandScript/blob/master/src/main/java/space/bbkr/sandscript/impl/ScriptedBlock.java#L49), but that runs into the issue of `NoSuchMethodException` being thrown if that method isn't impl'd.

Reply:
In the implementation that I tried, I just removed the script from execution when any errors have shown up. Printed the error to the console. If it errored, then it won't be executed again.

Reply:
Yeah, I should just include an API script after we figured how to export features. It should be using a registration method instead of the tag.

Reply:
OK